app/models/concerns/accountable.rb

```diff
@@ -0,0 +1,111 @@
+# frozen_string_literal: true
+
+module Accountable
```

Comment:
We need to migrate all models to use `has_account` instead of `belongs_to :account`.

The main benefits here are better validations, and account inheritance via `default:`.